src/system/interface.rs

```diff
@@ -160,39 +160,35 @@ mod test {
     use crate::system::{Group, User, ROOT_GROUP_NAME};
     use std::ffi::CString;
 
-    fn test_user(user: impl UnixUser, name_c: &CStr, uid: UserId) {
+    fn test_user(user: impl UnixUser, name_c: &CStr) {
         let name = name_c.to_str().unwrap();
         assert!(user.has_name(name));
-        assert!(user.has_uid(uid));
         if user.has_name("root") {
             assert!(user.in_group_by_name(CString::new(ROOT_GROUP_NAME).unwrap().as_c_str()));
         } else {
             assert!(user.in_group_by_name(name_c));
         }
+        assert_eq!(user.is_root(), user.has_uid(UserId::ROOT));
```

Comment:
This is a tautology. `is_root` is implemented using `.has_uid(UserId::ROOT)`. Maybe instead put the check in the `user.has_name("root")` branch?